### DISCUSSION TITLE
"Ending node must be either a Chain or Agent"

### DISCUSSION THREAD

**Discussion by KewkLW** [FlowiseAI/Flowise#928] [Q&A]
Completely stuck on this and believe I have tried connecting every node but it won't connect to anything and won't output. 

![image](https://github.com/FlowiseAI/Flowise/assets/57611539/9d4f3a12-388a-46c6-9e65-8d5604470bcd)

**Comment by KewkLW** [comment]
This is resolved, it stemmed from having a corrupt credential. I removed them all and then readded them which resolved the issue.

**Comment by JoeChildress** [comment]
I'm running into this. Do you have details on the credential issue? Is it an API key?

**Comment by stueynet** [comment]
Hmm throwing my hat in the ring as well. Deleting and generating and adding my credentials did not solve it for me. Could we re-open?

**Comment by shyamchandranmec** [comment]
I am facing the same issue. Please reopen

**Comment by stueynet** [comment]
What solved it for me was switching to a docker based installed.

**Comment by thanhnam46** [comment]
Same issue here, pls open!

**Reply by thanhnam46** [reply]
Instead of imstall locally, i used docker and it worked perfectly

**Comment by JoeChildress** [comment]
I think this was resolved when I downloaded the repo and ran it locally.
